File: bench/py/fib_typed.py
# Typed recursive Fibonacci - Python port of bench/qn/fib_typed.qn.
# Run: python3.13 bench/py/fib_typed.py
#
# Kept as a class with a method (not a bare function) to preserve the
# dispatch-per-call shape of the Quoin original's `.value:` send. The type
# annotations mirror the Quoin source's `|n: Integer ^Integer|`; Python does
# not act on them at runtime.


class Fib:
    def value(self, n: int) -> int:
        if n <= 1:
            return n
        return self.value(n - 1) + self.value(n - 2)


r = Fib().value(32)
if r == 2178309:
    print('fib_typed: ok')
else:
    print('fib_typed: FAIL got ' + str(r))

File: bench/rb/fib_typed.rb
# Typed recursive Fibonacci - in Quoin this exercises the typed/devirt hot
# path (sealed Integer arithmetic, typed params and return). Ruby has no type
# annotations, so this differs from fib_untyped.rb only in workload size
# (n=32); kept as a separate file so cross-language timings stay 1:1 with
# bench/qn/fib_typed.qn.
# Run: `ruby bench/rb/fib_typed.rb`.

class Fib
  def self.value(n)
    if n <= 1
      n
    else
      value(n - 1) + value(n - 2)
    end
  end
end

r = Fib.value(32)
if r == 2178309
  puts 'fib_typed: ok'
else
  puts "fib_typed: FAIL got #{r}"
end
